### PR TITLE
Fix PED bug for Consultation and Sessions

### LINKED ISSUE #171: [PE-D][Tester E] "takeattendance" command does not seem to work as intended when student is deleted
"takeattendance" command does not seem to work as intended when I have removed the student, "Alex Yeoh" in this screenshot below. 

In detail,
- Attendance on tab 4 (Sessions) does not show that the student has been deleted and does not show that the attendance of student is changed even when I executed the command to mark student as absent and present multiple times. 
- The output result could be more helpful in helping readers understand what went wrong. 

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: zannloo/ped#3

### LINKED ISSUE #180: [PE-D][Tester D] viewattendance user guide bug
The UG mentions that the student who's attendance you want to view must exist in FAKEJARVIS. However after deleting the student from FAKEJARVIS, you are still able to view the student's attendance. Can consider rephrasing this portion.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: samuelmui8/ped#5

### LINKED ISSUE #196: [PE-D][Tester C] Broken feature
Alex is not removed after running the command shown

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Jonyxzx/ped#9

### LINKED ISSUE #204: [PE-D][Tester A] UI Bug - Output shown to the user seems to be formatted without spaces between the student name, could potentially mislead users of the application
User 1:

User 2:

When doing `createconsult d/10/10/2023 tt/15:00 n/alexyeoh n/disheng`

I end up getting => 

And this might be a problem as alexyeohdisheng could very well be one person's name. Improving the output might help.

-------------
Labels: `type.FunctionalityBug` `severity.VeryLow`
original: AlagappanRa/ped#7

### LINKED ISSUE #205: [PE-D][Tester A] Integration Test - Edits does not update names various tabs.
Student 1:

Student 2:

Consulation set up using createsession command. 

Now, I proceed to edit the name of student 1 because i realise i mispelt his name so i change it to alexteo 

I end up getting this as the output.

Now when i scroll over to the consultation tab using tab 5

The consultation session between alexteo (after re-naming) and disheng still shows up as alexyeoh and disheng under the tags.

Same issue with sessions tabs.

TLDR:
Name does not update under consultations > tags and under sessions > tags when the name edit is made under students tab.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: AlagappanRa/ped#8

### LINKED ISSUE #225: Any `edit` comand replaces the old student object, causing commands downstream to not work as intended
Steps to replicate bug:

`editgradedtest INDEX PREFIX/SCORE`
** or any edit functions in general**

Subsequently, any command that references to the old `student object` such as `takeattendance` would not work as intended

### PR DESCRIPTION
**Pull request by wesho1107**
This PR aims to address the bugs issued during PED:
- Display of success message when using `createconsult`
- Display of updated consultation/session after any `edit` or `delete` features

### CODE REVIEW

File: src/main/java/seedu/address/model/person/StudentSet.java [hunk 1]
```diff
@@ -100,20 +107,28 @@ public String toStudentNames() {
         return studentNames.toString();
     }
 
-
     /**
      * Creates and returns a new SessionStudents object with the same students as this one.
      *
      * @return A new SessionStudents object with a copy of the students.
```
Comment (raydenlim) [inline src/main/java/seedu/address/model/person/StudentSet.java:113]: returns a new StudentSet

File: src/main/java/seedu/address/model/person/StudentSet.java [hunk 2]
```diff
@@ -26,7 +24,7 @@ public class SessionStudents implements Iterable<Person> {
     /**
      * Constructs an empty SessionStudents object.
```
Comment (raydenlim) [inline src/main/java/seedu/address/model/person/StudentSet.java]: Constructs an empty StudentSet

File: src/main/java/seedu/address/model/session/Session.java [hunk 3]
```diff
@@ -94,8 +95,25 @@ public void addStudent(Person student) {
      *
      * @param student The student to remove.
```
Comment (raydenlim) [inline src/main/java/seedu/address/model/session/Session.java]: add a `@return` Returns new and updated StudentSet object

File: src/main/java/seedu/address/model/consultation/ConsultationList.java [hunk 4]
```diff
@@ -54,6 +56,39 @@ public void remove(Consultation toRemove) {
         }
     }
 
+    /**
+     * Removes the {@code student} from all consultations in the consultation list.
```
Comment (raydenlim) [inline src/main/java/seedu/address/model/consultation/ConsultationList.java:60]: It only removes the student's instance from 1 consultation list right? not all

File: src/main/java/seedu/address/model/session/SessionList.java [hunk 5]
```diff
@@ -122,6 +123,39 @@ public void remove(Session toRemove) {
         }
     }
 
+    /**
+     * Removes the {@code student} from all sessions in the sessions list.
```
Comment (raydenlim) [inline src/main/java/seedu/address/model/session/SessionList.java:127]: Not sure if this removes student from ALL session, or just 1 session

**Review by raydenlim — APPROVED** [review]
Apart from some minor details,
LGTM!!

File: src/main/java/seedu/address/model/consultation/ConsultationList.java [hunk 6]
```diff
@@ -54,6 +56,39 @@ public void remove(Consultation toRemove) {
...
+    /**
+     * Replaces the student {@code target} in the list with {@code editedStudent}.
+     */
+    public void setStudent(Person target, Person editedStudent) {
+        requireAllNonNull(target, editedStudent);
+        List<Consultation> newConsultations = new ArrayList<>();
+
+        for (Consultation consultation : internalList) {
+            if (consultation.contains(target)) {
+                consultation.removeStudent(target);
```
Comment (Respirayson) [inline src/main/java/seedu/address/model/consultation/ConsultationList.java]: perhaps can abstract the 2 commands here into a replace command in Consultation

File: src/main/java/seedu/address/model/consultation/ConsultationList.java (hunk 4 above)
Comment (wesho1107) [reply src/main/java/seedu/address/model/consultation/ConsultationList.java:60]: Yep, but it's from all the consultations in that consultation list.

File: src/main/java/seedu/address/model/session/SessionList.java (hunk 5 above)
Comment (wesho1107) [reply src/main/java/seedu/address/model/session/SessionList.java:127]: It removes the student from all sessions in the sessions list. The idea is to map remove functions on each `Session` object so there is no more instance of the student in anywhere in the sessions of the list.

File: src/main/java/seedu/address/model/consultation/ConsultationList.java (hunk 6 above)
Comment (wesho1107) [reply src/main/java/seedu/address/model/consultation/ConsultationList.java]: Sounds like a good idea!

**Pull request merged**